src/server/ApiDispatchers.cpp

```diff
@@ -12,6 +12,18 @@
 #include "../host/telemetry.hpp"
 #include "../host/cmdline.h"
 
+// Assumes that it will find <m> in the calling environment.
+#define TraceConsoleAPICallWithOrigin(ApiName, ...)       \
+    TraceLoggingWrite(                                    \
+        g_hConhostV2EventTraceProvider,                   \
+        "API_" ApiName,                                   \
+        TraceLoggingPid([&] { const auto p=m->GetProcessHandle(); return p ? p->dwProcessId : 0; }(), "OriginatingProcess"), \

```

Comment:
These could be lifted out if reviewers desire.

Reply:
I personally would prefer that. This will produce a unique lambda instance per macro invocation which then might get folded into a single instance in Release builds if the compiler realizes they're all the same assembly. But I believe it's safer if we add free-standing `TraceGetProcessId` and `TraceGetThreadId` functions instead that take a `CONSOLE_API_MSG*` and put them above the macro. It would also make compilation a teeny tiny bit cheaper.

(We could also add them as member functions to `CONSOLE_API_MSG`, but I would greatly prefer if we moved `CONSOLE_API_MSG` back to being a trivial class. I'm really not sure if the "writing data off the end of the class" thing is legal C++ when the struct is non-trivial and I'd rather not tempt fate. 😓)

Reply:
I switched to functions. Trust me, the `CONSOLE_API_MSG` class gives me the ABSOLUTE willies